frontend/src/types/MetadataPair.ts

```diff
@@ -0,0 +1,4 @@
+export type MetadataPair = {
```

Comment:
It might be better to write this as a generic `KeyValuePair` type instead of domain binding it to metadata, as k/v can be reused later for tag as well for example. It might also be coerceable into a typescript Record<string, string> type, but not quite as familiar with this.

Reply:
I made `MetadataPair` because there was an existing `Tagging` type that used another `Tag` type which is a key value pair, and that was used throughout the application already.
So It was seeming wierd having one of them as a `Record<string, string>` and one a named `Tag` type.
I kind of agree that the generic record one makes more sense but didn't want to go through adjusting existing uses of `Tag`